asgiref/server.py

```diff
@@ -88,7 +88,7 @@ def get_or_create_application_instance(self, scope_id, scope):
         input_queue = asyncio.Queue()
         application_instance = guarantee_single_callable(self.application)
         # Run it, and stash the future for later checking
-        future = run_future(
+        future = asyncio.ensure_future(
```

Comment:
We actually want to keep `run_future` here as it uses the improve `asyncio.run` in 3.7 and above

Reply:
I admit it looks a bit to break the thing done in  compatibility.py, but  I think somehow ```asyncio.run``` fits not well when the server task is running endlessly with while loop.  
```asyncio.run```   cannot be called from a running event loop and it closes the loop while it completes the tasks.  

```
# cpython/Lib/asyncio/runners.py 
def run(main, *, debug=None):
    if events._get_running_loop() is not None:
        raise RuntimeError(
            "asyncio.run() cannot be called from a running event loop")

    loop = events.new_event_loop()
    try:
        events.set_event_loop(loop)
    #  the codes are omitted.
```
When a server task is inside a running loop,  the server class cannot call ```get_or_create_application_instance``` method due to  ```asyncio.run```.  I prefer to use ```asyncio.ensure_future``` here, and let people define when and how to close the loop by inherent class.